integration/insert_compat_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/AlekSi/pointer"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+)
+
+func TestInsertCompat(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]insertCompatTestCase{
+		"InsertEmptyDocument": {
+			insert:     bson.D{},
+			resultType: nonEmptyResult,
+		},
+		"InsertIDArray": {
+			insert:     bson.D{{"_id", bson.A{"foo", "bar"}}},
+			resultType: emptyResult,
+		},
+		"InsertIDRegex": {
+			insert:     bson.D{{"_id", "^regex$"}},
```

Comment:
I understand thats not a valid regex type syntax and that's just a basic string, but I noticed unexpected behavior. This operation is correct so the change is applied to collection, but every subtest inside this test cases says that this id is duplicated, which from my understanding shouldn't be true because  `SetupCompat` creates different collections for every subtest

Reply:
I think I understand :)

You have a loop `for _, doc := range allDocs {`, and it seems that you try to insert multiple times there. It was reasonable for update compat test where we might want to try updating every single value, but for `insert` it doesn't make sense. I'll leave a comment there.